### ISSUE TITLE
Build fails but error is not detected

### ISSUE THREAD

**Issue by feedhandler** [ja11sop/cuppa#41]
For some kinds of build failures, the error is not detected. 
E.g. we have a build failure due to issue #40 but it is not detected:
```
dmsdev@dmsdev> scons --offline  --cache-root=/_cuppa/_cache
scons: Reading SConscript files ...
cuppa: version: [info] cuppa: version 0.9.26.1
cuppa: core: [info] Running in OFFLINE mode
cuppa: core: [info] using sconstruct file [SConstruct]
cuppa: configure: [info] No settings to load, skipping configure
Could not detect the version of libc installed. You might be missing some development libraries!
cuppa: core: [info] available toolchains are ['gcc62', 'gcc', 'clang34', 'clang']
cuppa: core: [info] default profiles are ['devtoolset']
cuppa: core: [info] default dependencies are ['Core', 'boost', 'date', 'json', 'log4cpp', 'pugixml']
cuppa: core: [info] Using sub-sconscripts ['./src/SConscript']
cuppa: core: [info] Default variants of ['dbg'] being used.
cuppa: location: [info] Downloading [http://tpnexus.eur.ad.tullib.com:8081/repository/core-services/3rdparty/sources/boost_1_66_0.tar.gz]...
cuppa: location: [info] Download progress |====10%====20%====30%====40%====50%====60%====70%====80%====90%====| Complete
cuppa: location: [info] [http://tpnexus.eur.ad.tullib.com:8081/repository/core-services/3rdparty/sources/boost_1_66_0.tar.gz] successfully downloaded to [/tmp/tmpubEPZU.gz]
cuppa: registration: [error] init_env_for_variant in cuppa.methods.build_with.BuildWithMethod failed with error ['module' object has no attribute 'ContentTooShortError']
cuppa: __init__: [error] Cuppa terminated by exception [AttributeError: 'SConsEnvironment' object has no attribute 'BoostSharedLibs']
cuppa: __init__: [error] Use --verbosity=exception (or above) to see the stack
scons: done reading SConscript files.
scons: Building targets ...
scons: `.' is up to date.
scons: done building targets.
dmsdev@dmsdev>
```

**Comment by ja11sop** [comment]
Closed by incorrectly labelled commit so reopening

**Comment by ja11sop** [comment]
This should be fixed with the latest commit 790d4af
